packages/solana/src/index.ts

```diff
@@ -20,14 +21,51 @@ export class TurnkeySigner {
     const fromKey = new PublicKey(fromAddress);
     const messageToSign = tx.serializeMessage();
 
+    const signRawPayloadResult = await this.signRawPayload(
+      messageToSign.toString("hex"),
+      "PAYLOAD_ENCODING_HEXADECIMAL",
+      fromAddress
+    );
+
+    const signature = `${signRawPayloadResult.signRawPayloadResult?.r}${signRawPayloadResult.signRawPayloadResult?.s}`;
+
+    tx.addSignature(fromKey, Buffer.from(signature, "hex"));
+  }
+
+  /**
+   * This function takes a message and returns it after being signed with Turnkey
+   *
+   * @param message The message to sign (Uint8Array)
+   * @param fromAddress Solana address (base58 encoded)
+   */
+  public async signMessage(
+    message: Uint8Array,
+    fromAddress: string
+  ): Promise<Uint8Array> {
+    const signRawPayloadResult = await this.signRawPayload(
+      Buffer.from(message).toString("hex"),
+      "PAYLOAD_ENCODING_HEXADECIMAL",
+      fromAddress
+    );
+    return Buffer.from(
+      `${signRawPayloadResult.signRawPayloadResult?.r}${signRawPayloadResult.signRawPayloadResult?.s}`,
+      "hex"
+    );
+  }
+
+  private async signRawPayload(
+    payload: string,
+    encoding: definitions["v1PayloadEncoding"],
```

Comment:
You should be able to remove this now that both signMessage and addSignature use the same encoding 🎉